app/javascript/components/util/DynamicHeaders.js

```diff
@@ -0,0 +1,23 @@
+// To promote reusable and accessibile containers, this function will return
+
+function generateDynamicHeaders(priority) {
+  if (isNaN(priority) || priority === null || priority === undefined || priority < 1 || priority > 6) {
+    return null;
+  }
+  let headerMappings = [
+    { label: 'one', value: 1 },
+    { label: 'two', value: 2 },
+    { label: 'three', value: 3 },
+    { label: 'four', value: 4 },
+    { label: 'five', value: 5 },
+    { label: 'six', value: 6 },
+  ];
+
+  let headerArray = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];
+  let availableHeaders = headerArray.slice(priority - 1);
+  let headersObject = {};
+  headerMappings.forEach((hm, index) => (headersObject[`${hm.label}`] = availableHeaders[Number(index)] || null));
+  return headersObject;
```

Comment:
I think the dynamic headers can be accomplished with a simpler solution. If you define a function:
```javascript
const Heading = ({level, children, ...props}) => {
	return React.createElement(`h${Math.min(level, 6)}`, props , children)
}
```

Then you can use it like:
```javascript
constructor(props) {
  const rootHeaderLevel = props.headerLevel;
}
```
```html
<Heading level={rootHeaderLevel}>Heading X</Heading>
<Heading level={rootHeaderLevel + 1}>Heading Y</Heading>
<Heading level ={rootHeaderLevel + 2}>Heading Z</Heading>
```

This was derived from [here](https://jsfiddle.net/robstarbuck/g9hbdqne/39/).

Thoughts?

Reply:
This is very clever, and seems to work! I'll go about implementing this.